src/App.tsx

```diff
@@ -1,39 +1,68 @@
-import React from 'react';
+import { Component } from 'react';
+import { Clock } from './components/clock/Clock';
+
 import './App.scss';
 
-function getRandomName(): string {
+const getRandomName = (): string => {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
-}
+};
+
+type State = {
+  clockName: string,
+  hasClock: boolean,
+};
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component {
+  state: Readonly<State> = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  timerId = 0;
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.hideOnClick);
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.addEventListener('click', this.showOnClick);
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    this.timerId = window.setInterval((this.newClockName), 3300);
+  }
 
-        {' time is '}
+  componentWillUnmount() {
+    document.removeEventListener('click', this.showOnClick);
+    document.removeEventListener('contextmenu', this.hideOnClick);
+    window.clearInterval(this.timerId);
+  }
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+  newClockName = () => {
+    this.setState({
+      clockName: getRandomName(),
+    });
+  };
+
+  hideOnClick = (event:MouseEvent) => {
+    event.preventDefault();
+    if (event.button === 2) {
+      this.setState({
+        hasClock: false,
+      });
+    }
+  };
+
+  showOnClick = () => {
+    this.setState({
+      hasClock: true,
+    });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {this.state.hasClock && <Clock clockName={this.state.clockName} />}
```

Comment:
use desctructuring for getting needed properties from the state object before `return` statement